src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_3_horizontal_layout.html.twig

```diff
@@ -25,13 +25,18 @@ col-sm-2
 {# Rows #}
 
 {% block form_row -%}
-    <div class="form-group{% if (not compound or force_error|default(false)) and not valid %} has-error{% endif %}">
-        {{- form_label(form) -}}
-        <div class="{{ block('form_group_class') }}">
-            {{- form_widget(form) -}}
-            {{- form_errors(form) -}}
+    {% set displayMarkup = form.children|length == 0 or form.vars.label is not same as(false) %}
```

Comment:
I think it's a bug displaying a `form-group` inside a `form-group`. The label i still shown - the markup is not rendered when the label is set to false. 
I cannot see how this will be solved in your pr. It looks like another feature.